ceremonies/src/lib.rs

```diff
@@ -359,148 +414,326 @@ decl_error! {
         for cid in cids.iter() {
-            <ParticipantRegistry<T>>::remove_prefix((cid, cindex), None);
-            <ParticipantIndex<T>>::remove_prefix((cid, cindex), None);
-            <ParticipantCount>::insert((cid, cindex), 0);
+            <BootstrapperRegistry<T>>::remove_prefix((cid, cindex), None);
+            <BootstrapperIndex<T>>::remove_prefix((cid, cindex), None);
+            <BootstrapperCount>::insert((cid, cindex), 0);
+
+            <ReputableRegistry<T>>::remove_prefix((cid, cindex), None);
+            <ReputableIndex<T>>::remove_prefix((cid, cindex), None);
+            <ReputableCount>::insert((cid, cindex), 0);
+
+            <EndorseeRegistry<T>>::remove_prefix((cid, cindex), None);
+            <EndorseeIndex<T>>::remove_prefix((cid, cindex), None);
+            <EndorseeCount>::insert((cid, cindex), 0);
+
+            <NewbieRegistry<T>>::remove_prefix((cid, cindex), None);
+            <NewbieIndex<T>>::remove_prefix((cid, cindex), None);
+            <NewbieCount>::insert((cid, cindex), 0);
+
+
+            <AssignedBootstrapperCount>::insert((cid, cindex), 0);
+            <AssignedReputableCount>::insert((cid, cindex), 0);
+            <AssignedEndorseeCount>::insert((cid, cindex), 0);
+            <AssignedNewbieCount>::insert((cid, cindex), 0);
+
+            AssignmentParamsBootstrappersReputables::remove((cid, cindex));
+            AssignmentParamsEndorsees::remove((cid, cindex));
+            AssignmentParamsNewbies::remove((cid, cindex));
+
+
             <Endorsees<T>>::remove_prefix((cid, cindex), None);
-            <MeetupRegistry<T>>::remove_prefix((cid, cindex), None);
-            <MeetupIndex<T>>::remove_prefix((cid, cindex), None);
             <MeetupCount>::insert((cid, cindex), 0);
             <AttestationRegistry<T>>::remove_prefix((cid, cindex), None);
             <AttestationIndex<T>>::remove_prefix((cid, cindex), None);
             <AttestationCount>::insert((cid, cindex), 0);
             <MeetupParticipantCountVote<T>>::remove_prefix((cid, cindex), None);
+
+            <IssuedRewards>::remove_prefix((cid, cindex), None);
         }
         debug!(target: LOG, "purged registry for ceremony {}", cindex);
     }
 
-    // this function is expensive, so it should later be processed off-chain within SubstraTEE-worker
-    // currently the complexity is O(n) where n is the number of registered participants
-    fn assign_meetups() {
+    fn generate_meetup_assignment_params(community_ceremony: CommunityCeremony) -> DispatchResult {
+        let meetup_multiplier = 10u64;
+        let num_locations = <encointer_communities::Module<T>>::get_locations(&community_ceremony.0).len() as u64;
+        let num_boostrappers = Self::bootstrapper_count(community_ceremony);
+        let num_reputables = Self::reputable_count(community_ceremony);
+        let num_endorsees = Self::endorsee_count(community_ceremony);
+        let num_newbies = Self::newbie_count(community_ceremony);
```

Comment:
moving the creation of a variable towards the first use allows the reader of code to not fill his brain at once.. also it shows more visibly that (e.g. num_locations) certain variables are used only once and can then be "forgotten"

also eliminating local variables is sometimes desirable: maybe num_locations could be eliminated by directly matching the result of get_locations.. only, if the code becomes more readable